[WIP] feat: support underline for markdown and update string format pattern

Bad3r opened the pull request:
Update formatting functions and format-text! to use %s pattern

TODO

- [ ] use `util/get-selected-text`

Depends on:
- #9388

breaking change:
- Remove `/underline` command and replace it with `modKey + u` shortcut

issues:
- close #9390
- close https://github.com/logseq/logseq/issues/8790#issuecomment-1464049503
- close #8819 (duplicate)

forum posts:
- https://discuss.logseq.com/t/more-options-for-text-formatting-bold-strikethrough-underline-etc/
- 


# QA Testing Instructions for Text Formatting Feature
Repeat the steps for markdown and org-mode formats.

## Initial Setup:
1. Create a new block with the sentence: `LOGSEQ: Logical Organization Graphical-System for Efficient Querying.`

## Test for Text Formatting Features:

### Bold:
2. Select the word `Graphical` and apply bold formatting (`$MOD+b`)
3. Expected Result: `**Graphical**|-System`
4. Verify the result and cursor (`|`) position.
Select the styled word `Graphical` again and remove the formatting by pressing the same shortcut.

### Italic:
5. Select the word `Graphical` and apply italic formatting (`$MOD+i`)
6. Expected Result: `*Graphical*|-System`
7. Verify the result and cursor (`|`) position.
Select the styled word `Graphical` again and remove the formatting by pressing the same shortcut.

### Underline:
8. Select the word `Graphical` and apply underline formatting (`$MOD+u`)
9. Expected Result: `<u>Graphical</u>|-System`
10. Verify the result and cursor (`|`) position.
Select the styled word `Graphical` again and remove the formatting by pressing the same shortcut.

### Strikethrough:
11. Select the word `Graphical` and apply the strikethrough formatting (`$MOD+Shift+s`).
12. Expected Result: `~~Graphical~~|-System`
13. Verify the result and cursor (`|`) position.
Select the styled word `Graphical` again and remove the formatting by pressing the same shortcut.

### Sequential Application and Removal of Formatting:
14. Select the word `Graphical` and apply all the text formatting shortcuts. Example: `$MOD+i`, `$MOD+u`, `$MOD+Shift+s`, `$MOD+b`.
15. Expected Result: `**~~<u>*Graphical*</u>~~**|-System`
16. Verify the result and cursor (`|`) position.
17. Then, remove each formatting one by one in reverse order.
18. Expected Result: `Graphical|-System` (`Graphical` should remain selected)
19. Expected Result: The word `Graphical` should remain selected after removing the formatting.

## Test for Formatting Without Selection:
26. In a new block without any text selected, apply each formatting command or shortcut. Example: `$MOD+i`, `$MOD+u`, `$MOD+Shift+s`, `$MOD+b`.
15. Expected Result: `**~~<u>*|*</u>~~**`
28. Verify the result and cursor position.

Please remember to document any discrepancies between the expected and actual results, along with any error messages or issues encountered during testing.



https://user-images.githubusercontent.com/25513724/235405226-f17ee2ab-3f55-4a34-85bb-a35e91324d08.mp4


This PR updates the text formatting functions to use a consistent %s pattern for wrapping text with the appropriate formatting markers. The `format-text!` function has also been updated to accommodate these changes.

- Updated get-bold, get-italic, get-strike-through, get-underline, and get-highlight functions to use %s pattern
- Modified format-text! to handle the new %s pattern in the formatting functions
- Ensured compatibility with existing text formatting behavior

The result of this change is introducing a new feature; underline text using `mod + u`. Since markdown doesn't support underlining, HTML syntax is needed. Due to the opening and closing tags being different for HTML, there was a need to update the `text-format!` function to accommodate this case. Instead of writing a special case for underlines, I opted to modify the existing format used to a more malleable one.










---

> **Warning**
> Old PR description

Discord: https://discord.com/channels/725182569297215569/735747000649252894/1100463176270745640

This pull request introduces a new feature that adds a keyboard shortcut for underlining text `Mod + u` in both Markdown and Org-mode formats. The implementation modifies the `get-underline` function and introduces the `underline-format!` function to support the respective underline syntax for each format. In Org-mode, the underscore (_) syntax is used, while in Markdown, the text is wrapped with the `<u>` and `</u>` HTML tags.

Since Markdown does not natively support underlines, I opted to use HTML tags as a workaround. This presented a challenge as the opening and closing tags are different. Initially, I considered modifying the `format-text!` function to handle this case, but doing so could potentially break other features. As a result, I decided to not change it.

This approach may have introduced some code duplication, and there could be a more efficient solution beyond my current understanding of the code. I apologize if this adds extra work to the review process. Please let me know if you have any suggestions for improvements or if further modifications are needed.

All translations have been added and its been added to the shortcuts page.

Video Demo:

https://user-images.githubusercontent.com/25513724/234464747-0310092b-6c94-424b-a65d-f3486ee69b5b.mp4


QA Steps:
- open a markdown and/or org-mode graph
- select some text
- use the shortcut Mod + u to underline it

Test unwrapping
- select wrapped text
- use the same shortcut to undo the formatting.

Bad3r committed:
enhance: Add shortcut to underline text

Bad3r committed:
enhance: Add underline to dict

Bad3r committed:
enhance: Add underline for MD mode using HTML syntax

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
enhance: Use similar syntax for consistency

Bad3r committed:
Revert "enhance: Use similar syntax for consistency"

This reverts commit f85d380ad9afd6f1103157da6f17e9761b59902e.

Bad3r committed:
enhance: Use similar syntax for consistency

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
enhance: Update formatting functions and format-text! to use %s pattern
This commit updates the text formatting functions to use a consistent %s pattern for wrapping text with the appropriate formatting markers. The format-text! function has also been updated to accommodate these changes.

- Updated get-bold, get-italic, get-strike-through, get-underline, and get-highlight functions to use %s pattern
- Modified format-text! to handle the new %s pattern in the formatting functions
- Ensured compatibility with existing text formatting behavior

Bad3r commented:
@logseq/core, this PR is ready for review. Not sure who to ask? The original code is written by @tiensonqin ~3 years ago. Is this change in pattern OK?

Bad3r committed:
fix: set default value

Bad3r committed:
enhance: standardize the use of 'strikethrough' for better readability and consistency

Bad3r committed:
fix: org-mode italic format string

Bad3r committed:
ehnance: improve format-text! logic

Bad3r committed:
ehnance: improve formatting

Bad3r committed:
ehnance: improve formatting

Bad3r committed:
Revert "ehnance: improve formatting"

This reverts commit 3ddff8f6d92ebb3d4c760a9fbb49f698ed10b7c5.

Bad3r committed:
Revert "ehnance: improve formatting"

This reverts commit 3ddff8f6d92ebb3d4c760a9fbb49f698ed10b7c5.

Bad3r commented:
setting it as WIP for now as it could be improved to be more efficient.

Bad3r committed:
fix: unwrapping and cursor postion

Bad3r committed:
fix: unwrapping and cursor postion

Bad3r committed:
fix: remove redundancy

Bad3r committed:
fix: remove redundancy

Bad3r committed:
enhance: streamline the code

Bad3r committed:
make the code more idiomatic

Bad3r committed:
make the code more idiomatic

Bad3r committed:
Delete config.edn

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
Delete config.edn

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
Fix: unwrapping

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
fix: cursor possiton

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
@logseq/core This PR is ready for review :eyes:

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
When wrapping is removed should text selection remain or should it be cleared? :thinking:

Bad3r committed:
enhance: org-mode ~strikethrough~ & _underline_

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
clean up 🧹

Bad3r committed:
fix: cursor possiton calc

Bad3r committed:
enhance: don't clear selection when unwrapping

Bad3r committed:
enhance: Use descriptive naming

Bad3r committed:
enhance: Use descriptive naming

Bad3r committed:
clean up 🧹

Bad3r committed:
clean up 🧹

Bad3r committed:
fix: cursor possiton calc

Bad3r committed:
fix: selection calc

Bad3r committed:
fix: selection calc

Bad3r committed:
enhance: clear selection after wapping text

clstb committed:
fix: preserve properties on update-block

Bad3r committed:
enhance: more idiomatic code

clstb committed:
feat: add test for save-block

Bad3r committed:
use + for strikethrough in org-mode instead of ~

candideu committed:
Add custom.js script to html

candideu committed:
Export custom.js

Bad3r commented:
@andelf PTAL when you get a chance :)

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Trigger testing 🧪

Bad3r commented:
Please add the 'looking for review' label

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

clstb committed:
feat: improve test comment

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

logseq-cldwalker committed:
Add standard lint tasks to common lib

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

logseq-cldwalker committed:
Add :hidden config support to graph-parser.cli

Also fix lints and tests from previous commits

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

cnrpman commented:
Is there any hint for doing QA?
And I suggest adding cljs tests and E2E coverage, as Editor features are heavily coupled 👀

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
> Sorry for the late reply. I didn't notice this PR till now. Is there any hint for doing QA?

No worries! I updated the original post with instructions.

> 1. I suggest adding cljs tests and E2E coverage, as Editor features are heavily coupled eyes
>    And I saw a bunch of helper functions here:
>    https://github.com/logseq/logseq/pull/9245/files#diff-2f055ca9c790c38c4cbe601146ca7c19ad359f36a73554039581c2e5fae428d6R131-R175
> ...
> Also, can you check if there's any logic that can reuse?

Only one is reused here. I tried to make the functions I wrote as reusable as possible. I looked into the codebase and logic the only function I am reusing is `reset-cursor-range`. I could merge the code into fewer functions which is what I initially had, but I thought this was better for readability and reusability. I'll review it again to make sure I didn't miss anything.

For testing, I had hard time getting the E2E test going. See the discord discussion:
```md
I attempted writing some tests for it but there isn't a current similar e2e test. I did do rigorous manual testing. I am not familiar with how playwright works. If someone from the logseq team could give me a pointer or write one test case for any styling and then I could adapt it to write tests for the other cases.

here is an example test case:
- Create a block with a sentence for example: `LOGSEQ: Logical Organization Graphical-System for Efficient Querying.`
- Select the word `Graphical`
  - make it bold `**Graphical**`
    - check if cursor is moved after the styling i.e. `**Graphical**<cursor>-System`
  - Select the word `Graphical` again to remove the formatting
    - expected output: `Graphical<cursor>-System`
    - confirm that the word `Graphical` is still selected
    - verify cursor position
```
https://discord.com/channels/725182569297215569/775937033394323478/1104095602696003606
I am new to this, so I think it will take me a long time to figure out how to do it correctly. An example would be great, but I understand that none currently exist that are similar, and check for the cursor position. cljs will probably be easier to do.


>    Hopefully can have some doc strings for better codebase maintaining

Will do I also tried to make the variables and function names as clear as possible.

> 2. Seems we are using `<ins></ins>` tags for markdown underline. Reference: https://docs.logseq.com/#/page/64526c5b-999d-4d09-9156-0ec00c65222f and https://github.com/logseq/logseq/blob/86103a47e5847da8263bad83f3e0784fd35414c4/src/main/frontend/commands.cljs#L231-L233

`ins` is not used for underlining in HTML:
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ins
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/u

Based on the git blame, this code was written by community contributors; maybe it wasn't reviewed carefully? I am not sure where/how that is even used; I'll look into it.


> Also a reminder of some items to do when this PR is ready to be merged :)
> 
> * prepare PR for docs
> * finding out all GitHub issues and forum threads with the underline feature shortcut request

Sounds good will make sure to do that.

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
@cnrpman, only the underline in markdown has the command `/underline`; as mentioned, it doesn't use the recommended HTML underline syntax. This command does not work for orgmode, and there are no similar commands for bold, strikethrough, or italic. I think that command should be removed.

Bad3r commented:
@cnrpman, only the underline in markdown has the command `/underline`; as mentioned, it doesn't use the recommended HTML underline syntax. This command does not work for orgmode, and there are no similar commands for bold, strikethrough, or italic. I think that command should be removed.

Bad3r committed:
enhance: Remove /underline command

Bad3r committed:
enhance: Remove unused func state/markdown?

Bad3r commented:
> @Bader I see. It sounds like a workaround towards the missing of the underline shortcut in that time? It's good to me to remove it since we're introducing shortcut support (but still it's a breaking change worth mention)
> > I did try to remove it but linting returned an error related to large-vars do to state/markdown? this method is only used here
> I don't see a lint error in the CI. Is this resolved?

Yes it was a workaround before the shortcut existed. Also, yes I removed both the command and the `state/markdown?` method to get the linting test to pass.

I am trying to get an E2E test going for this PR.

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

avelino committed:
front/i18n: isolate package languages

Signed-off-by: Avelino <avelinorun@gmail.com>

avelino committed:
shortcut/i18n: isolate package languages

Signed-off-by: Avelino <avelinorun@gmail.com>

avelino committed:
update documentation to new translation structure (standalone package)

Signed-off-by: Avelino <avelinorun@gmail.com>

avelino committed:
centralize translation dictionary

Signed-off-by: Avelino <avelinorun@gmail.com>

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
enhance: use '_' for italic to solve #8790

Bad3r committed:
enhance: only autopair tilda when text is selected

Bad3r committed:
feat: autopair quotation marks

- also auto pair % when a text is selected for example %APPDATA%

Bad3r committed:
Enhance/editor string formatting e2e (#48)

* enhance(E2E): Test editor italic, bold, strikethrough and underline.

* disable underline tests until #9245 is merged

* Fix: position

* enhance: add test with all formatting

* enhance: add tests for editor autopairing

* enhance: add tests for editor autopairing

* add: underline test as fixme as a prereq to #9245

* enhance: break tests into steps

* revert: test steps

---------

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
Enhance/editor string formatting e2e (#48)

* enhance(E2E): Test editor italic, bold, strikethrough and underline.

* disable underline tests until #9245 is merged

* Fix: position

* enhance: add test with all formatting

* enhance: add tests for editor autopairing

* enhance: add tests for editor autopairing

* add: underline test as fixme as a prereq to #9245

* enhance: break tests into steps

* revert: test steps

---------

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
enable tests

Bad3r committed:
Merge 'enhance/editor-string-formatting-e2e' into enhance/underline-shortcut

Bad3r committed:
fix: cursor position calculation

Bad3r committed:
update editor tests

Bad3r committed:
update editor tests

Bad3r committed:
revert style changes

Signed-off-by: Bad3r <bad3r@protonmail.com>

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
use underscore for italic

Bad3r committed:
fix: cursor possiton calculation

Bad3r committed:
fix test

tiensonqin committed:
fix: broken block ref when backspace/delete a block

xyhp915 committed:
enhance(ui): align the number list with the bullet list prefix marker

xyhp915 committed:
enhance(ui): align the number list with the bullet list prefix marker

tiensonqin committed:
fix: transaction

tiensonqin committed:
remove unnecessary save-current-block

tiensonqin committed:
fix: can't jump to the correct position when editing number list

andelf committed:
enhance(editor): restore scroll position new adding newline in a block

tiensonqin committed:
fix: handle the case when both blocks are referenced

andelf committed:
enhance(editor): restore scroll position new paste in a long block

tiensonqin committed:
fix: cursor not jump back when undo a DELETE at the end action

It also remove built-in properties such as id from the editor content

xyhp915 committed:
fix(mobile): incorrect padding between the bullet marker and the content boundary on mobile

tiensonqin committed:
comment on setTimeout in undo

xyhp915 committed:
fix(ui): incorrect position of the number list fold arrow on mobile

tiensonqin committed:
fix: replace with source block's content when deleting refed blocks

tiensonqin committed:
fix: undo should bring refs back

logseq-cldwalker committed:
Split out graph fns and reuse them b/n gp-cli and electron

Also added test

tiensonqin committed:
fix: replace empty target when paste cut blocks

tiensonqin committed:
fix: lint

tiensonqin committed:
chore: uncomment code

stdword committed:
docs: update templates/config.edn

typos & spaces

Bad3r committed:
Update config.edn

sprocketc committed:
fix multiselect right click on converted shape

sprocketc committed:
fix: undo/redo of converted shapes

sprocketc committed:
fix: context bar actions of converted shapes

tiensonqin committed:
add e2e tests

sprocketc committed:
chore: add tests

sprocketc committed:
revert unneeded change

logseq-cldwalker committed:
Fix current block with disappearing path refs - close #9365

outliner-core/save-block and save-block-inner! were retracting
block/path-refs aggressively, especially for cases when no path-refs
had changed

logseq-cldwalker committed:
Add a test to ensure there's no regression on bug

Also introduce helper fn for common db fixture setup

tiensonqin committed:
fix: util/mobile? is a function

andelf committed:
refactor(editor): add nearest-scrollable-container util fn

cnrpman committed:
Enhance: version control settings description update (#9373)

* enhance: remove version control entry blocker

* ui: version control text box update

Co-authored-by: Gabriel Horner <97210743+logseq-cldwalker@users.noreply.github.com>

---------

Co-authored-by: Gabriel Horner <97210743+logseq-cldwalker@users.noreply.github.com>

sprocketc committed:
revert mobile? fn

sprocketc committed:
chore: add and fix e2e tests

sprocketc committed:
fix: test description

sprocketc committed:
style: run prettier

andelf committed:
fix(fs): add unicode normalize to path-fn

This is a catch-all type fix

andelf committed:
fix(fs): use normalized relative path for mobile platform

Fix #9378

logseq-cldwalker committed:
Document repl test workflow as a followup to #9372

Also organized unit test sections more

avelino committed:
shortcut/config: use default dicts (EN)

Signed-off-by: Avelino <avelinorun@gmail.com>

avelino committed:
docs/lang: describe how to add a new language

Signed-off-by: Avelino <avelinorun@gmail.com>

logseq-cldwalker committed:
test refactor

Bring back helper and cleanup lets

Bad3r committed:
dev(playwright): Ignore conntection refused errors (#9426)

* ignore connection refused errors

* enhance: ignore connection errors

andelf committed:
fix(editor): show block bullet when a blank block has children

RCmerci committed:
enhance(export): add title when page opml-export

logseq-cldwalker committed:
Fix app not loading correctly and most CI failures

-Most frontend.modules.shortcut.core requires shouldn't have been
changed
- Revert back pt-BR and ru langs which had changes caught by bb lang:list
- Fix typos CI failing by updating config

tiensonqin committed:
Changes suggested from Zhiyuan

logseq-cldwalker committed:
Refactor and simplify dicts

- frontend.dicts encapsulates dicts behavior for all other namespaces
- Each dict ns only has one var which prepares us
- No longer need shortcut.dicts
- Fix remaining lints that were broken
- Also bring back frontend.dicts
- Update guide
- Add a false binding to :editor/toggle-undo-redo-mode which throws a
  needless warning

hasecilu committed:
Update Spanish translations with new format

andelf committed:
fix(editor): auto complete parenthesis at line start

Michele0303 committed:
Update it.cljc

andelf committed:
chore(mobile): drop support for ios 13

logseq-cldwalker committed:
Fix parse-graph not ignoring correctly for relative dir

parse-graph now returns absolute file paths

giuseppedandrea committed:
enhance: calculator command on a non-blank line

candideu committed:
Add test for custom.js to export_test.cljs

candideu committed:
Add test for custom.js to export_test.cljs

Fixed missing thing

andelf committed:
fix(android): apply url decode before fs watcher notifies

tiensonqin committed:
Update src/main/frontend/state.cljs

Co-authored-by: Gabriel Horner <97210743+logseq-cldwalker@users.noreply.github.com>

tiensonqin committed:
Update src/main/frontend/modules/outliner/datascript.cljc

Co-authored-by: Gabriel Horner <97210743+logseq-cldwalker@users.noreply.github.com>

tiensonqin committed:
fix: parens

tiensonqin committed:
Move cut-paste? logic to paste handler

tiensonqin committed:
fix: relax same block id check only when deleting a source block

tiensonqin committed:
fix: remove ugly timeout

andelf committed:
refactor(ios): new share intent implementation

logseq-cldwalker committed:
custom.js should load last in publishing app

tiensonqin committed:
Update comment

dependabot[bot] committed:
chore(deps): bump posthog-js from 1.10.2 to 1.57.2

Bumps [posthog-js](https://github.com/PostHog/posthog-js) from 1.10.2 to 1.57.2.
- [Release notes](https://github.com/PostHog/posthog-js/releases)
- [Changelog](https://github.com/PostHog/posthog-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/PostHog/posthog-js/compare/v1.10.2...v1.57.2)

---
updated-dependencies:
- dependency-name: posthog-js
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

andelf committed:
fix(ios): no handling of uri pasting

Fix #9433

andelf committed:
fix(ios): wrong link when sharing asset link (#9462)

Fix #9454

optimistic5 committed:
main/frontend/dicts/ru: prepare translation, add all elements

optimistic5 committed:
main/frontend/dicts/ru: update translation

optimistic5 committed:
main/frontend/dicts/ru: update cycle-todo

andelf committed:
chore(release): bump version 0.9.7

giuseppedandrea committed:
fix: scroll the view when selecting multiple blocks (#9444)

andelf committed:
Revert "chore(deps): bump posthog-js from 1.10.2 to 1.57.2"

This reverts commit 1ce4697b22f396ea72ddeecf4d688dcafadea5e2.

andelf committed:
chore(release): bump version 0.9.8

Bad3r committed:
enhance(E2E): editor string formatting and symbols autopairing tests (#9388)

* enhance: only autopair tilda when text is selected

* Enhance/editor string formatting e2e (#48)

* enhance(E2E): Test editor italic, bold, strikethrough and underline.

...

---------

Signed-off-by: Bad3r <bad3r@protonmail.com>

andelf committed:
ci: run e2e when releasing (#9478)

Bad3r committed:
dev: enable E2E tests for underline

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/logseq/logseq?pullRequest=9245) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/logseq/logseq?pullRequest=9245) before we can accept your contribution.<br/>**11** out of **12** committers have signed the CLA.<br/><br/>:white_check_mark: RCmerci<br/>:white_check_mark: andelf<br/>:white_check_mark: logseq-cldwalker<br/>:white_check_mark: Michele0303<br/>:white_check_mark: hasecilu<br/>:white_check_mark: giuseppedandrea<br/>:white_check_mark: avelino<br/>:white_check_mark: tiensonqin<br/>:white_check_mark: optimistic5<br/>:white_check_mark: Bad3r<br/>:white_check_mark: candideu<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/logseq/logseq?pullRequest=9245) it.</sub>

Bad3r committed:
fix: merge conflict

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented on src/main/frontend/state.cljs:
```diff
@@ -433,10 +433,10 @@ should be done through this fn in order to get global config and config defaults
-  []
-  (= (keyword (get-preferred-format))
-     :markdown))
+;; (defn markdown?
```
No longer used

cnrpman commented:
Humm, seems tons of conflicts given the great changes of the dicts

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
> Humm, seems tons of conflicts given the great changes of the dicts Also, the commit history is so awful. Better to do a thorough interactive rebase from the current master

@cnrpman fixed the conflict. This PR is on hold until #9391 is merged.

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r committed:
Merge branch 'master' into enhance/underline-shortcut

Bad3r commented:
Close due to team not having time to review community PRs and to save my time.
